### ISSUE TITLE
"device offline" blocked connection error when using Hyper-V virtual switch on Windows

### ISSUE THREAD

**Issue by novacrazy** [mullvad/mullvadvpn-app#2153]
## Issue report

Operating system: Windows 10 Pro for Workstations version 1909 build 18363.1082

App version: 2020.5 (latest stable as of writing this)

### Issue description

In order to allow virtual machines raw access to the network, I use a virtual switch in Hyper-V feeding directly to the ethernet adapter. The "real" adapter is setup as a Hyper-V switch, so IPv4/6 is disabled on it. 

I used to be able to use Mullvad with this setup, but weird things happened and despite writing this issue on the same machine, now Mullvad cannot detect the correct adapter to connect through. It can access account details just fine, it's only when it attempts to make any OpenVPN/Wireguard connect that it fails.

Furthermore, the reference Wireguard client for Windows using Mullvad config files is able to connect just fine with my setup, so it's most definitely something in the Mullvad Client.

My adapter setup is laid out as such:

`Ethernet 3`: Main external Intel(R) Gigabit adapter, setup as Hyper-V switch. IPv4/6 connectivity is disabled on this.
`vEthernet (Default switch)`: Default internal (inter-VM) adapter
`vEthernet (Plex 2)`: Hyper-V adapter that connects directly to Ethernet 3, IPv4/6 is enabled on this, and both the host OS and VMs can access it.
`Mullvad`: VPN TAP adapter (only used for OpenVPN)
`mullvad-wg`: Mullvad Wireguard adapter, only exists when connected to WG.

Furthermore, while troubleshooting this, attempting to enable IPv4/6 on Ethernet 3 will allow Mullvad to establish connections, but will block all traffic to `vEthernet (Plex 2)`, including in the virtual machines.

What's more odd is that this used to work up until about two weeks ago when some weird intermittent issues with my ISP connection caused multiple deadlocks in Mullvad/DNS/Ethernet/etc. and required multiple adapter/settings resets.

**Comment by mvd-ows** [comment]
After a failed connection attempt, what does the error contained in `C:\ProgramData\Mullvad VPN\daemon.log` say, and is there any extra information being logged?

**Comment by mvd-ows** [comment]
Oh wait, did the app not give you the option to connect because it thinks there are no usable network adapters?

**Comment by novacrazy** [comment]
Well, this is awkward. The logs were not updating as I tried to connect multiple times, so in an attempt to get a fresh/clean log for you, I removed the old logs and restarted the daemon process (just terminated it). When the daemon and the app started back up, it connected automatically with no issues.

One thing I did change was the option to allow "local network sharing", which solved it blocking traffic between the host OS and virtual machines. Disabling that broke the VMs again, but the VPN itself was still able to connect. 

I apologize for such a dumb mistake, deleting the old logs, but now we know that forcibly restarting the daemon can force a refresh of some kind that apparently multiple system reboots cannot. 

In that case, feel free to close this for now.
